cmd/otel-allocator/config/config.go

```diff
@@ -39,18 +39,20 @@ const DefaultConfigFilePath string = "/conf/targetallocator.yaml"
 const DefaultCRScrapeInterval model.Duration = model.Duration(time.Second * 30)
 
 type Config struct {
-	ListenAddr             string             `yaml:"listen_addr,omitempty"`
-	KubeConfigFilePath     string             `yaml:"kube_config_file_path,omitempty"`
-	ClusterConfig          *rest.Config       `yaml:"-"`
-	RootLogger             logr.Logger        `yaml:"-"`
-	ReloadConfig           bool               `yaml:"-"`
-	LabelSelector          map[string]string  `yaml:"label_selector,omitempty"`
-	PromConfig             *promconfig.Config `yaml:"config"`
-	AllocationStrategy     *string            `yaml:"allocation_strategy,omitempty"`
-	FilterStrategy         *string            `yaml:"filter_strategy,omitempty"`
-	PrometheusCR           PrometheusCRConfig `yaml:"prometheus_cr,omitempty"`
-	PodMonitorSelector     map[string]string  `yaml:"pod_monitor_selector,omitempty"`
-	ServiceMonitorSelector map[string]string  `yaml:"service_monitor_selector,omitempty"`
+	ListenAddr                      string             `yaml:"listen_addr,omitempty"`
+	KubeConfigFilePath              string             `yaml:"kube_config_file_path,omitempty"`
+	ClusterConfig                   *rest.Config       `yaml:"-"`
+	RootLogger                      logr.Logger        `yaml:"-"`
+	ReloadConfig                    bool               `yaml:"-"`
+	LabelSelector                   map[string]string  `yaml:"label_selector,omitempty"`
+	PromConfig                      *promconfig.Config `yaml:"config"`
+	AllocationStrategy              *string            `yaml:"allocation_strategy,omitempty"`
+	FilterStrategy                  *string            `yaml:"filter_strategy,omitempty"`
+	PrometheusCR                    PrometheusCRConfig `yaml:"prometheus_cr,omitempty"`
+	PodMonitorSelector              map[string]string  `yaml:"pod_monitor_selector,omitempty"`
+	ServiceMonitorSelector          map[string]string  `yaml:"service_monitor_selector,omitempty"`
+	ServiceMonitorNamespaceSelector map[string]string  `yaml:"service_monitor_namespace_selector,omitempty"`
+	PodMonitorNamespaceSelector     map[string]string  `yaml:"pod_monitor_namespace_selector,omitempty"`
```

Comment:
Thanks @jaronoff97. Since the common [prometheus fields](https://github.com/open-telemetry/opentelemetry-operator/pull/2328/files#:~:text=%7D%2C-,ServiceMonitorNamespaceSelector%3A%20%26metav1.LabelSelector%7B,%7D%2C,-%7D%2C) have the ServiceMonitorNamespaceSelector and PodMonitorNamespaceSelector already initialized, even if the targetallocator config is not setting these selectors (or have nil selectors), the default behavior will be to match all namespaces which is the current behavior. But I can add defaults, in case you think otherwise.  Please let me know.